operations/ftdc.go

```diff
@@ -737,19 +737,89 @@ func toT2() cli.Command {
 			ctx, cancel := context.WithCancel(context.Background())
 			defer cancel()
 
-			inputFile, err := os.Open(inputPath)
+			inputStat, err := os.Stat(inputPath)
 			if err != nil {
 				return errors.Wrapf(err, "problem opening file '%s'", inputPath)
 			}
-			defer func() { grip.Warning(inputFile.Close()) }()
 
-			// All genny output files are named using the workload actor and operation.
-			//   i.e., Actor.Operation.ftdc
-			//
-			// We get the actor and operation names from the ftdc filepath for use
-			// in the translation process.
-			fileName := filepath.Base(inputPath)
-			actorOperation := strings.Split(fileName, ".ftdc")
+			// Stores ftdc chunk iterators and its metadata to be used in TranslateGenny.
+			var outputSlice []*ftdc.GennyOutputMetadata
+
+			// Check if input path is an individual file or directory.
+			// All ftdc files in the provided input directory are initially
+			// used to set GennyOutputMetadata Name, Iter, StartTime and EndTime.
+			switch mode := inputStat.Mode(); {
+			// Handle directory.
+			case mode.IsDir():
+				input, err := os.Open(inputPath)
+				if err != nil {
+					return errors.Wrapf(err, "problem opening input path '%s'", inputPath)
+				}
+				defer func() { grip.Warning((input.Close())) }()
+
+				files, err := input.Readdir(-1)
+				if err != nil {
+					return errors.Wrapf(err, "problem reading dir '%s'", inputPath)
+				}
+
+				for _, file := range files {
+					if file.Mode().IsRegular() && filepath.Ext(file.Name()) == ".ftdc" {
+						var gennyOutput ftdc.GennyOutputMetadata
+						filePath := filepath.Join(inputPath, file.Name())
+
+						f, err := os.Open(filePath)
+						if err != nil {
+							return errors.Wrapf(err, "problem opening file '%s'", filePath)
+						}
+						gennyOutput.Iter = ftdc.ReadChunks(ctx, f)
+
+						// GetGennyTime exhausts the current chunk iterator and renders it
+						// unusable for future tasks.
+						gennyOutput = ftdc.GetGennyTime(ctx, gennyOutput)
+						err = f.Close()
+						if err != nil {
+							return errors.Wrapf(err, "problem opening file '%s'", filePath)
+						}
+
+						// Reopen the file to get a new chunk iterator for the ftdc file.
+						f, err = os.Open(filePath)
```

Comment:
You don't close this file.

Reply:
I explained in a prior comment chain
> I couldn't call input.Close on L817 because TranslateGenny seems to rely on the files remaining open otherwise this error throws:
> runtime error: invalid memory address or nil pointer dereference

Should I store these open files in a slice and close them after TranslateGenny?

Reply:
Does `TranslateGenny` close the file reader? If so, then you do not need to do anything, otherwise use a defer statement to close it.